### PR TITLE
Hoc

### PR DESCRIPTION
**Pull request by josepplloo**
## Summary

- hoc added.

## Screenshots

hoc

## Trello Card

https://trello.com/c/DMfTomzB/19-patterns-ii-hocs

### CODE REVIEW

**Review by CrisVega08 — CHANGES_REQUESTED** [review]
Jose this pr has conflicts solve please.

File: react/tic-tac-toe/src/app/components/AppRouter/components/AuthInfo/styles.module.scss [hunk 1]
```diff
@@ -2,42 +2,109 @@
...
+  &::after {
+    border-top: 30px solid $black;
+    border-left: 1px solid transparent;
+    border-right: 10px solid transparent;
+    content: '';
+    margin: 0 0 10px 0;
+    position: absolute;
+    right: 0px;
+    top: 13px;
+    width: 91px;
```
Comment (CrisVega08) [inline react/tic-tac-toe/src/app/components/AppRouter/components/AuthInfo/styles.module.scss]: 90px

File: react/tic-tac-toe/src/app/components/AppRouter/components/AuthInfo/styles.module.scss [hunk 2]
```diff
@@ -2,42 +2,109 @@
...
+    position: absolute;
+    width: 91px;
+  }
+
+  &::after {
+    border-top: 30px solid $black;
+    border-left: 1px solid transparent;
+    border-right: 10px solid transparent;
+    content: '';
+    margin: 0 0 10px 0;
```
Comment (CrisVega08) [inline react/tic-tac-toe/src/app/components/AppRouter/components/AuthInfo/styles.module.scss]: margin-botom: 10px;

File: react/tic-tac-toe/src/redux/matches/reducer.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import { actions } from './actions';
+
+const initialStateDescription = {
+  matches: []
+};
+
+const initialState = completeState(initialStateDescription);
+
+const reducerDescription = {
+  primaryActions: [actions.MATCHES]
```
Comment (CrisVega08) [inline react/tic-tac-toe/src/redux/matches/reducer.js]: Beautiful ❤️

File: react/tic-tac-toe/src/redux/matches/actions.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { createTypes, completeTypes } from 'redux-recompose';
+
+import MatchesServices from '~services/MatchesService';
+
+export const actions = createTypes(completeTypes(['MATCHES']), '@@MATCHES');
```
Comment (CrisVega08) [inline react/tic-tac-toe/src/redux/matches/actions.js]: 👏

File: react/tic-tac-toe/src/redux/game/actions.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+export const actions = createTypes(completeTypes(['WINNER'], ['ON_CLICK', 'ON_MOVE']), '@@GAME');
+
+const actionsCreator = {
+  winner: values => ({
+    type: actions.WINNER,
+    target: 'game',
+    service: MatchesService.postMatch,
+    payload: values
+  }),
+  onClick: (id, history, stepNumber, xIsNext) => dispatch => {
```
Comment (CrisVega08) [inline react/tic-tac-toe/src/redux/game/actions.js]: This don't like, the onClick method have more mean on component

File: react/tic-tac-toe/src/app/components/AppRouter/components/AuthInfo/styles.module.scss [hunk 6]
```diff
@@ -2,42 +2,109 @@
...
 .error {
-  color: orange;
+  background-color: $orange;
+  border-radius: 2px;
+  color: $white;
+  margin-bottom: 10px;
+  padding: 5px 100px;
+  position: relative;
+
+  &::before {
```
Comment (CrisVega08) [inline react/tic-tac-toe/src/app/components/AppRouter/components/AuthInfo/styles.module.scss]: almost every psuedo elements have similar properties use a extends something like 
```%base-psuedo { ... } ```

File: react/tic-tac-toe/src/app/components/AppRouter/index.js [hunk 7]
```diff
@@ -1,4 +1,5 @@
 import React, { Component, Fragment } from 'react';
+import Spinner from 'react-spinkit';
```
Comment (CrisVega08) [inline react/tic-tac-toe/src/app/components/AppRouter/index.js]: The main idea for loader is made create a HOC that wrapper the component not put a spinner. 💀 💀 💀 💀

File: react/tic-tac-toe/src/app/screens/Game/index.js [hunk 8]
```diff
@@ -9,57 +13,29 @@ import Moves from './components/Moves';
...
-    history: [
-      {
-        squares: Array(9).fill(null)
-      }
-    ],
-    xIsNext: true,
-    stepNumber: 0
-  };
-
-  calculateWinner = squares => {
```
Comment (CrisVega08) [inline react/tic-tac-toe/src/app/screens/Game/index.js]: If prefer you use this here I only send the action when the winner exist.

File: react/tic-tac-toe/src/app/screens/Game/styles.module.scss [hunk 9]
```diff
@@ -8,7 +8,7 @@
     border-image: linear-gradient(to bottom, $black, transparent) 1 100%;
     border-style: solid;
     border-width: 3px;
-    color: white;
+    color: $white;
     max-width: 250px;
     padding: 1rem;
```
Comment (CrisVega08) [inline react/tic-tac-toe/src/app/screens/Game/styles.module.scss]: we usually don't use rem try with prixels

File: react/tic-tac-toe/src/app/screens/Podium/index.js [hunk 10]
```diff
@@ -1,30 +1,25 @@
...
 
-import MatchesService from '~services/MatchesService';
+import { PLAYER_ONE, PLAYER_TWO } from '~constants/';
+
+import matchesActions from '~redux/matches/actions';
 
 import styles from './styles.module.scss';
 
 class Podium extends Component {
-  state = {
```
Comment (CrisVega08) [inline react/tic-tac-toe/src/app/screens/Podium/index.js]: 👏

**Review by CrisVega08 — CHANGES_REQUESTED** [review]

File: react/tic-tac-toe/src/app/components/AppRouter/components/AuthInfo/styles.module.scss [hunk 11]
```diff
@@ -1,43 +1,111 @@
...
+    
+    bottom: 18px;
+  }
+
+  &::after {
+    @extend %title-spoiler;
+    
+    right: 0px;
+    transform: rotate(180deg);
+    top: 28px;
```
Comment (CrisVega08) [inline react/tic-tac-toe/src/app/components/AppRouter/components/AuthInfo/styles.module.scss]: order properties

File: react/tic-tac-toe/src/app/components/AppRouter/index.js [hunk 12]
```diff
@@ -63,7 +63,8 @@ AppRouter.propTypes = {
 
 const mapStateToProps = store => ({
   onError: store.auth.isloginError,
-  isAuth: store.auth.islogin
+  isAuth: store.auth.islogin,
+  onLoading: store.auth.isloginLoading
```
Comment (CrisVega08) [inline react/tic-tac-toe/src/app/components/AppRouter/index.js]: Delete it prop.

File: react/tic-tac-toe/src/app/components/hocs/index.js [hunk 13]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import Spinner from 'react-spinkit';
+
+function higherOrderComponent(WrappedComponent) {
```
Comment (CrisVega08) [inline react/tic-tac-toe/src/app/components/hocs/index.js]: Change this name for something that describe function.

File: react/tic-tac-toe/src/app/components/hocs/index.js [hunk 14]
```diff
@@ -0,0 +1,10 @@
...
+  return function HOC({ onLoading, ...props }) {
```
Comment (CrisVega08) [inline react/tic-tac-toe/src/app/components/hocs/index.js]: Same here

File: react/tic-tac-toe/src/app/screens/Game/components/Moves/index.js [hunk 15]
```diff
@@ -18,9 +18,9 @@ class Moves extends Component {
 }
 
 Moves.propTypes = {
-  desc: string.isRequired,
-  move: number.isRequired,
-  onClick: func.isRequired
+  onClick: func.isRequired,
+  desc: string,
```
Comment (CrisVega08) [inline react/tic-tac-toe/src/app/screens/Game/components/Moves/index.js]: desc and move are required?

File: react/tic-tac-toe/src/app/screens/Game/components/Square/index.js [hunk 16]
```diff
@@ -20,9 +20,9 @@ class Square extends Component {
 }
 
 Square.propTypes = {
-  id: number.isRequired,
-  value: string.isRequired,
-  onClick: func.isRequired
+  onClick: func.isRequired,
+  id: number,
```
Comment (CrisVega08) [inline react/tic-tac-toe/src/app/screens/Game/components/Square/index.js]: Same here

File: react/tic-tac-toe/src/app/screens/Podium/index.js [hunk 17]
```diff
@@ -1,52 +1,49 @@
 import React, { Component } from 'react';
-import Spinner from 'react-spinkit';
+import { connect } from 'react-redux';
+import { bool, func, arrayOf, string, shape, number } from 'prop-types';
 
-import MatchesService from '~services/MatchesService';
+import { PLAYER_ONE, PLAYER_TWO } from '~constants/';
```
Comment (CrisVega08) [inline react/tic-tac-toe/src/app/screens/Podium/index.js]: If you remove props these constants never use

**Review by CrisVega08 — CHANGES_REQUESTED** [review]

File: react/tic-tac-toe/src/app/screens/Game/components/Moves/index.js (hunk 15 above)
Comment (josepplloo) [reply react/tic-tac-toe/src/app/screens/Game/components/Moves/index.js]: only move.

File: react/tic-tac-toe/src/app/screens/Game/components/Square/index.js (hunk 16 above)
Comment (josepplloo) [reply react/tic-tac-toe/src/app/screens/Game/components/Square/index.js]: only id

File: react/tic-tac-toe/src/app/screens/Podium/index.js (hunk 17 above)
Comment (josepplloo) [reply react/tic-tac-toe/src/app/screens/Podium/index.js]: It will use for avoid linter problems with snake case.

**Review by CrisVega08 — APPROVED** [review]

**Pull request merged**